tests/test_modeling_blenderbot.py

```diff
@@ -0,0 +1,213 @@
+        
+        batch_size = input_ids.size(0)
+        config = BlenderbotConfig(hidden_size=16,vocab_size=self.vocab_size, encoder_layers=2, decoder_layers=2,
+                                  encoder_attention_heads=2,
+                                  decoder_attention_heads=2,
+                                  encoder_ffn_dim=8,
+                                  decoder_ffn_dim=8,
+                                  max_position_embeddings=24,
+                                  eos_token_id=2,
+                                  pad_token_id=0,
+                                  bos_token_id=1)
+        return config, input_ids, mask, batch_size
+    def test_generation_with_labels_forward(self):
+        config, input_ids, mask, batch_size = self.get_config_data()
+        labels = input_ids
+        model = BlenderbotConditionalGeneration(config)
+        model.to(torch_device)
+        outputs = model(input_ids=input_ids,
+                        attention_mask=mask,
+                        decoder_input_ids=input_ids,
+                        labels=labels)
+        loss = outputs[0]
+        self.assertIsInstance(loss.item(), float)   
+        scores = outputs[1]
+        
+        expected_shape = (batch_size, input_ids.size(1), self.vocab_size)
+        self.assertEqual(scores.size(),expected_shape)
+        
+    def test_generate_beam_search(self):
+        config, input_ids, mask, batch_size = self.get_config_data()
+       
+        model = BlenderbotConditionalGeneration(config)
+        model.eval()
+        max_seq_len = 10
+        generated_ids = model.generate(
+                input_ids.clone(),
+                do_sample=True,
+                num_return_sequences=1,
+                num_beams=2,
+                no_repeat_ngram_size=3,
+                max_length=max_seq_len
+        )
+        self.assertTrue(generated_ids.size(), (batch_size, max_seq_len))    
+     
+    @unittest.skipIf(torch_device == "cpu", "half precision can't be used on cpu")    
+    def test_model_fp16_forward(self):
+        config, input_ids, mask, batch_size = self.get_config_data()
+        model = BlenderbotConditionalGeneration(config=config)
+        model.to(torch_device)
+        model.half()
+        model.eval()
+        output = model(input_ids, attention_mask=mask)[0]
+        self.assertFalse(torch.isnan(output).any().item())
+
+    @unittest.skipIf(torch_device == "cpu", "half precision can't be used on cpu")
+    def test_model_fp16_generate(self, config, input_ids, input_mask):
+        config, input_ids, mask, batch_size = self.get_config_data()
+        model = BlenderbotConditionalGeneration(config=config)
+        model.to(torch_device)
+        model.half()
+        model.eval()
+        output = model.generate(input_ids, attention_mask=mask, do_sample=False)
+        self.assertFalse(torch.isnan(output).any().item())
+        
+    @slow
+    def test_inference(self):
```

Comment:
I think this test should be a little more demanding.
At the moment it just tests output shapes.
I think, once the tokenizer is figured out, it should test that for a reasonable (small) conversational history, a reasonable utterance is generated.

Reply:
100 % agree with @sshleifer here. We definitely need more integration tests that make sure you get 1-to-1 the same output result as the original model. For this you also don't need the tokenizer yet. 

Bart has a lot of these tests which I think are super important to be certain that the model works the way we expect it too work:
https://github.com/huggingface/transformers/blob/e33929ef1e98a71ba7fc411f39cbb5451396ef02/tests/test_modeling_bart.py#L234
and https://github.com/huggingface/transformers/blob/e33929ef1e98a71ba7fc411f39cbb5451396ef02/tests/test_modeling_bart.py#L557

Reply:
`test_xsum_summarization_same_as_fairseq` is probably the closest one to emulate.